src/cli/onefuzz/cred_wrapper.py

```diff
@@ -0,0 +1,66 @@
+# Wrap credentials from azure-identity to be compatible with SDK that needs msrestazure or azure.common.credentials
+# Need msrest >= 0.6.0
+# See also https://pypi.org/project/azure-identity/
+
+from typing import Any
+
+from azure.core.pipeline import PipelineContext, PipelineRequest
+from azure.core.pipeline.policies import BearerTokenCredentialPolicy
+from azure.core.pipeline.transport import HttpRequest
+from azure.identity import DefaultAzureCredential
+from msrest.authentication import BasicTokenAuthentication
+
+
+class CredentialWrapper(BasicTokenAuthentication):
+    def __init__(
+        self,
+        credential: Any = None,
+        resource_id: Any = "https://management.azure.com/.default",
+        **kwargs: Any
+    ):
+        """Wrap any azure-identity credential to work with SDK that needs azure.common.credentials/msrestazure.
+
+        Default resource is ARM (syntax of endpoint v2)
+
+        :param credential: Any azure-identity credential (DefaultAzureCredential by default)
+        :param str resource_id: The scope to use to get the token (default ARM)
+        """
+        super(CredentialWrapper, self).__init__({})
```

Comment:
In Python 3, we can just do `super().__init__({})`.

See [here](https://docs.python.org/3/library/functions.html#super).

Reply:
this is not my code. I do not write this, but we need to check it in (I included comment in the PR description with the link to Microsoft documentation)

see here
https://docs.microsoft.com/en-us/azure/developer/python/azure-sdk-authenticate-hosted-applications

Reply:
Not our code. This is required due to AppInsights module using older auth (and there is no newer AppInsights)

see https://docs.microsoft.com/en-us/azure/developer/python/azure-sdk-authenticate-hosted-applications

![image](https://user-images.githubusercontent.com/7307755/147987301-cb9a34bb-0192-4548-a95a-91a69a22c5ef.png)